command/ripe_atlas/traceroute.go

```diff
@@ -0,0 +1,167 @@
+package ripe_atlas
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/innogames/slack-bot/v2/bot"
+	"github.com/innogames/slack-bot/v2/bot/matcher"
+	"github.com/innogames/slack-bot/v2/bot/msg"
+	"github.com/innogames/slack-bot/v2/client"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/slack-go/slack"
+	"io"
+	"net/http"
+	"net/netip"
+	"time"
+)
+
+type tracerouteCommand struct {
+	bot.BaseCommand
+	cfg Config
+}
+
+func (c *tracerouteCommand) GetMatcher() matcher.Matcher {
+	return matcher.NewGroupMatcher(
+		matcher.NewRegexpMatcher(`traceroute (?P<TGT>.*)`, c.traceroute),
+	)
+}
+
+func (c *tracerouteCommand) traceroute(match matcher.Result, message msg.Message) {
+	destination := match.GetString("TGT")
+
+	c.AddReaction(":stopwatch:", message)
+	defer c.RemoveReaction(":stopwatch:", message)
+
+	var af int
+	address, err := netip.ParseAddr(destination)
+	if err != nil {
+		af = 6
+	} else {
+		if address.Is4() {
+			af = 4
+		} else {
+			af = 6
+		}
+	}
+
+	jsonData, _ := json.Marshal(MeasurementRequest{
+		Definitions: []MeasurementDefinition{
+			{
+				Af:             af,
+				Target:         destination,
+				Description:    fmt.Sprintf("Slackbot measurement to %s", destination),
+				Type:           "traceroute",
+				Protocol:       "ICMP",
+				Packets:        3,
+				ResolveOnProbe: false,
+				Paris:          0,
+				IsPublic:       true,
+			},
+		},
+		Probes: []Probes{
+			{
+				Type:      "area",
+				Value:     "WW",
+				Requested: 1,
+			},
+		},
+		IsOneOff: true,
+	})
+
+	url := fmt.Sprintf("%s/measurements", c.cfg.APIURL)
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(jsonData))
+	if err != nil {
+		c.ReplyError(message, errors.Wrap(err, "Request creation returned an err"))
+		return
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", "Key "+c.cfg.APIKey)
+
+	response, err := client.GetHTTPClient().Do(req)
+	if err != nil {
+		c.ReplyError(message, errors.Wrap(err, "HTTP Client Error"))
+		return
+	}
+	defer response.Body.Close()
+
+	if response.StatusCode >= 300 {
```

Comment:
Hmm 3xx codes are not errors, they are redirects.
Maybe you could try to centralise some of the request making / checking logic because some parts are very similar.